tests/leaderboards/test_tasks_leaderboard.py

```diff
@@ -0,0 +1,469 @@
+        observed (Dict[str, Any]): The observed dictionary to compare.
+        config_name (str): The name of the config to display in the error message.
+        module_name (str): The associated name of the subresults to display in the error message.
+        recursive (bool): Whether to compare nested structures.
+
+    Example:
+        reference = {
+            "transformers_version": "4.41.1",
+            "system_instruction_sha": None,
+            "versions": {"drop": 3.0}  # Nested structure, compared only if recursive is set to True
+        }
+        observed = {
+            "transformers_version": "4.41.1",
+            "system_instruction_sha": None,
+            "versions": {"drop": 3.0}  # Nested structure, compared only if recursive is set to True
+        }
+        compare_results(reference, observed, "example_config", "example_module")
+    """
+    if not reference:
+        raise ValueError("Reference dictionary is empty.")
+
+    for key, reference_val in reference.items():
+        if isinstance(reference_val, Dict):
+            if recursive:
+                assert key in observed, (
+                    f"Config: '{config_name} - {module_name}' failed. "
+                    f"Key '{key}' is missing in the observed dictionary."
+                )
+                compare_results(
+                    reference_val, observed[key], config_name, key, recursive
+                )
+        else:
+            assert key in observed, (
+                f"Config: '{config_name} - {module_name}' failed. "
+                f"Key '{key}' is missing in the observed dictionary."
+            )
+            observed_val = observed[key]
+            if isinstance(reference_val, float):
+                assert reference_val == pytest.approx(observed_val, abs=1e-4), (
+                    f"Config: '{config_name} - {module_name}' failed. {key}: "
+                    f"Expected: {repr(reference_val)}, got: {repr(observed_val)}"
+                )
+            else:
+                assert reference_val == observed_val, (
+                    f"Config: '{config_name} - {module_name}' failed. {key}: "
+                    f"Expected: {repr(reference_val)}, got: {repr(observed_val)}"
+                )
+
+
+@pytest.fixture(scope="module", params=load_all_configs(TEST_DEVICE))
+def load_config(request):
+    """
+    Pytest fixture that loads and yields evaluation configurations with expected results.
+
+    Use the `load_all_configs` function to retrieve configurations based on
+    the device specified in the `TESTS_DEVICE` environment variable. If the variable is
+    not set, it defaults to "cpu".
+
+    Args:
+        request: Pytest request object containing the parameter from the fixture decoration.
+
+    Yields:
+        Dict: A configuration dictionary containing:
+            - Evaluation settings
+            - Expected results for the given configuration
+    """
+    return request.param
```

Comment:
`load_all_configs` returns a list, doesn't this function return a list as well ? Or does pytest automatically yields each element of the list ?